Fix Ads Merchant Linking Failure Display Issue & Improve Retry Mechanism

ianlin opened the pull request:
### Changes proposed in this Pull Request:

<!-- You can erase any parts of this template not applicable to your Pull Request. -->

Closes #2834.

This PR fixes an issue where the merchant linking step fails but incorrectly displays as "connected," allowing users to proceed with onboarding. This ultimately results in a "Resource not found" error when attempting to create a campaign. Additionally, this PR improves the retry mechanism when fetching the merchant link API.

**Changes**

- Add one more condition to determine link merchant failure
  - When `hasAceess` is `true` but `status` is `incomplete` and `step` is `link_merchant`.
- Increase retry attempts to 5 times when fetching the merchant link API.
- Add a 1 second delay between each retry to improve API stability.

### Screenshots:

https://github.com/user-attachments/assets/dc145cdf-dcc9-4ea8-953f-7016870e2fa0

### Detailed test instructions:
<!-- Add detailed instructions for how to test that this PR fixes the issue and confirm that it doesn't break any other features :) -->

1. Go through the onboarding process.
2. Create a new ads account. It will attempt to link a merchant account.
3. If linking fails, verify that it does not display as "connected", and ensure the Continue button is disabled.
    - To make the linking fails, try to revert the commit 85477379ef6610a90648f2d2a2179ec7d3d03815 that increase the retries and add a delay
    - Note that it happens frequently but not every time, so try again if the retries without delay passed
5. If linking succeeds, proceed to the campaign creation step without encountering the "Resource not found" error.

### Additional details

**Update:** I have solved the problem and attached a video above.

~I initially wanted to include screentshots or videos in the PR, but my account ended up reaching the account creation limit like below screenshot. I then created a new Google account for testing, but it still showed account creation limit reached.~

![Screenshot 2025-04-02 at 11 27 56](https://github.com/user-attachments/assets/66e4cd72-c194-4b6e-8d06-fccc2aa45f24)

<!--
Optional.
Enter a summary of all changes in this Pull Request, which will be added to the changelog if accepted.
Each line should start with change type prefix`(Fix|Add|…) - `, for example:
> Break - A change breaking previous API or functionality.
> Add - A new feature, function or functionality was added.
> Update - Big changes to something that wasn't broken.
> Fix - Took care of something that wasn't working.
> Tweak - Small change, that isn't actually very important.
> Dev - Developer-facing only change.
> Doc - Updated customer or developer facing documentation

If you remove the "Changelog entry" header, the Pull Request title will be used as the changelog entry.

Add the `changelog: none` label if no changelog entry is needed.
-->
### Changelog entry

> Fix - Ads merchant linking failure display issue & improve retry mechanism

ianlin committed:
Prevent the user from proceeding when the link merchant step failed but hasAccess is true

ianlin committed:
Change retries to 5 and have a 1 seconde delay when getting merchant link

ianlin committed:
Fix typo

eason9487 committed:
Add a new state to the `useGoogleAdsAccountReady` hook to indicate whether it has linked to a Merchant Center account.

eason9487 committed:
Add a new state to the `useGoogleMCAccount` hook to indicate whether it has linked to an Ads account.

eason9487 committed:
Adjust the checks of onboarding and ads-onboarding flows to ensure the interlinking between Google Ads and Google Merchant Center accounts is completed.

eason9487 commented on js/src/hooks/useGoogleAdsAccountReady.js:
```diff
@@ -25,7 +27,14 @@ const useGoogleAdsAccountReady = () => {
 		return null;
 	}
 
+	// A temporary fix to prevent the user from proceeding when the link merchant step failed but hasAccess is true.
+	const linkMerchantFailed =
+		hasAccess &&
+		step === 'link_merchant' &&
+		googleAdsAccount?.status === GOOGLE_ADS_ACCOUNT_STATUS.INCOMPLETE;
```
It may cause the continue button to be disabled when a user connects to Ads first then Merchant Center, and the two "✅ Connected" labels are not shown.

![1](https://github.com/user-attachments/assets/3f892982-09c9-4eab-9064-54fa5cb2e8cb)

eason9487 reviewed (commented):
Thanks for the fix! From my testing, the merchant linking failure no longer occurs.

I will push alternative solution later to resolve the issue in the `useGoogleAdsAccountReady` hook.

eason9487 committed:
Make Copilot reviewer happy.

Address: https://github.com/woocommerce/google-listings-and-ads/pull/2835#pullrequestreview-2742587173

eason9487 committed:
Add a code comment to clarify why it uses the `isLinkedToMerchantCenter` state in the ads-onboarding flow.

Address: https://github.com/woocommerce/google-listings-and-ads/pull/2835#discussion_r2028596055

eason9487 commented on js/src/pages/ads-onboarding/ads-stepper/setup-accounts.js:
```diff
@@ -24,13 +24,13 @@ const SetupAccounts = ( props ) => {
 	}
 
-	const isContinueButtonDisabled = ! isGoogleAdsReady;
+	const isContinueButtonDisabled = ! isLinkedToMerchantCenter;
 
```
Added a code comment in https://github.com/woocommerce/google-listings-and-ads/pull/2835/commits/8c0a01274c12f1dcfc9a1d32e9af26cdaeac0fde.
